Packs/Box/Integrations/BoxV2/BoxV2.py

```diff
@@ -0,0 +1,2025 @@
+    return fileResult(filename=file_name, data=response.content)
+
+
+''' MAIN FUNCTION '''
+
+
+def test_module(client: Client) -> str:
+    """
+    This test assumes that the account has at least one user. By definition, if the instance is
+    configured correctly, there will always be a minimum of one user.
+
+    :param client:
+    :return:
+    """
+    response: Response = client.list_users(limit=1)
+    if response:
+        return 'ok'
+    else:
+        return 'An error occurred.'
+
+
+def fetch_incidents(client: Client, max_results: int, last_run: dict, first_fetch_time: int,
+                    as_user: str) -> Tuple[str, List[Dict]]:
+    """
+
+    :param client:
+    :param max_results:
+    :param last_run:
+    :param first_fetch_time:
+    :param as_user:
+    :return:
+    """
+    created_after = last_run.get('time', None)
+    incidents = []
+    if not created_after:
+        created_after = datetime.fromtimestamp(first_fetch_time, tz=timezone.utc).strftime(
+            DATE_FORMAT)
+    results = client.list_events(stream_type='admin_logs', as_user=as_user, limit=max_results,
+                                 created_after=created_after)
+    raw_incidents = results.get('entries', [])
+    next_run = datetime.now(tz=timezone.utc).strftime(DATE_FORMAT)
+    for raw_incident in raw_incidents:
+        event = Event(raw_input=raw_incident)
+        xsoar_incident = event.format_incident()
+        incidents.append(xsoar_incident)
+        if event.created_at > created_after:
+            next_run = event.created_at
+
+    return next_run, incidents
+
+
+def main() -> None:
+    """main function, parses params and runs command functions
+
+    :return:
+    :rtype:
+    """
+    # get the service API url
+    base_url = urljoin('https://api.box.com', '2.0')
+    verify_certificate = not demisto.params().get('insecure', False)
+
+    # Determine first fetch time if none is found.
+    first_fetch_time = arg_to_timestamp(
+        arg=demisto.params().get('first_fetch', '1 day'),
+        arg_name='First fetch time',
+        required=True
+    )
+    # Using assert as a type guard (since first_fetch_time is always an int when required=True)
+    assert isinstance(first_fetch_time, int)
+    proxy = demisto.params().get('proxy', False)
+    demisto.debug(f'Command being called is {demisto.command()}')
+    try:
+        client = Client(
+            auth_params=demisto.params(),
+            base_url=base_url,
+            verify=verify_certificate,
+            proxy=proxy)
+        if not any(command in demisto.command() for command in ['test-module', 'fetch-incidents',
+                                                                'box-find-file-folder-by-share-link']):
+            # Both commands should only use the explicit user they were assigned and not inherit
+            # the default user. This ensures as-user is used every time.
+            handle_default_user(args=demisto.args(), params=demisto.params())
```

Comment:
this may save some lines of code, but manipulating arguments and params creates a confusing code.
lets talk about a better solution here.